### ISSUE TITLE
Protect Eureka with user pass

### ISSUE THREAD

**Issue by n0daft** [ElderByte-/spring-eureka-server#1]
The aim is to secure the registration/deletion of services.

As of May 15, spring has no concrete solution for this: 

> Eureka Server can be secured using Spring Security. We don't have anything auto-configured. Securing eureka is currently an exercise for the developers. (See [here](http://stackoverflow.com/questions/30147541/securing-microservices-that-use-spring-cloud-netflixs-eureka-and-feign))

**Possible solutions**

*Spring Security*
Example for securing it with spring security can be found on [Stack Overflow](http://stackoverflow.com/questions/28974752/securing-eureka-in-spring-cloud).

**Comment by n0daft** [comment]
I've secured the eureka server with spring security. If pull request is merged, one **needs** to define the two environment variables **EUREKA_USER** and **EUREKA_PASS** on the eureka server. This is because I was no able to define default values for the spring security variables. As long as the other services do not specifically state other credentials in their environment, these values should be set to *admin* and *1337*. This is because I've set the default values in the other services to those values. However, to make it more flexible, these credentials are also configurable via environment
